client/src/App.tsx

```diff
@@ -39,27 +54,26 @@ export function App() {
     return () => {
       clearInterval(timer);
     };
-  }, []);
-
-  useEffect(() => {
-    if (ready) {
-      window.location.href = 'http://localhost:9890/';
-    }
-  }, [ready]);
+  }, [started, ready]);
 
   return (
-    <Grid container flex={1} direction="column" spacing={4}>
-      <Grid item justifyContent="center" textAlign="center" minHeight="80px">
-        {!ready && (
-          <>
-            <LinearProgress />
+    <>
+      {!ready && (
+        <Grid container flex={1} direction="column" padding="16px 32px" height="100%" justifyContent="center" alignItems="center">
+          <Grid item>
+            <LinearProgress/>
             <Typography mt={2}>
               Waiting for sqlcl to be ready. It may take some seconds if
               it's the first time.
             </Typography>
-          </>
-        )}
-      </Grid>
-    </Grid>
+          </Grid>
+        </Grid>
+      )}
+      {ready && (
+        <Box display="flex" flex={1} width="100%" height="100%">
+          <iframe src='http://localhost:9890/' width="100%" height="100%" />
```

Comment:
I changed the extension's behaviour to use an iframe rather than redirecting to the ttyd URL. This allows the theme to switch because the user is still in the React app. If redirected, the user is not in the react app anymore. Thus, nothing would happen when switching from dark to light mode.